prov/efa/src/efa_domain.c

```diff
@@ -230,6 +238,15 @@ int efa_domain_open(struct fid_fabric *fabric_fid, struct fi_info *info,
 
 	*domain_fid = &domain->util_domain.domain_fid;
 
+#ifdef HAVE_GDRCOPY
+	domain->gdr = ofi_gdrcopy_open();
```

Comment:
Shouldn't the gdrcopy open/close calls be in the hmem init/cleanup functions? You can add a function to cuda hmem so we can check for gdrcopy in the EFA provider.

Reply:
gdrcopy_open allocate a resource: gdr, which is essentially a file handle. I think it should be inside efa_domain. If we put it in hmem_gdrcopy_init, then it would be a global variable, and we will need a lock.

Reply:
I'm not sure how that's any different from the global handles already in the CUDA HMEM code. The ZE HMEM code stores global command queues as well.

Reply:
I was thinking about multi thread cases, put gdr in domain would allow each thread to have its own gdr resource. If we declare ```gdr``` as global variable, then all threads must use same ```gdr```. I am not familiar with the ZE interface, maybe the ZE routines operates on command queue are thread safe?

Reply:
@iziemba had exact same suggestion when reviewing https://github.com/ofiwg/libfabric/pull/6218, so I implement it as you both suggested.